crates/miden-testing/tests/scripts/faucet.rs

```diff
@@ -138,9 +150,224 @@ fn faucet_contract_mint_fungible_asset_fails_exceeds_max_supply() -> anyhow::Res
+                push.0.0.0 padw
+
+                push.{recipient}
+                push.{note_execution_hint}
+                push.{note_type}
+                push.{aux}
+                push.{tag}
+                push.{amount}
+                # => [amount, tag, aux, note_type, execution_hint, RECIPIENT, pad(7)]
+
+                call.::miden::contracts::faucets::basic_fungible::distribute
+                # => [note_idx, pad(15)]
+
+                # truncate the stack
+                dropw dropw dropw dropw
+            end
+            ",
+        note_type = note_type as u8,
+        recipient = recipient,
+        aux = aux,
+        tag = u32::from(tag),
+        note_execution_hint = Felt::from(note_execution_hint)
+    );
+
+    let mint_tx_script = ScriptBuilder::default().compile_tx_script(mint_tx_script_code)?;
+    let mint_tx_context = mock_chain
+        .build_tx_context(faucet.id(), &[], &[])?
+        .tx_script(mint_tx_script)
+        .build()?;
+    let mint_executed_transaction = mint_tx_context.execute_blocking()?;
+
+    prove_and_verify_transaction(mint_executed_transaction.clone())?;
+
+    let _minted_note = mint_executed_transaction.output_notes().get_note(0).clone();
+
+    // Now burn the minted asset
+    let note_script = "
+        # burn the asset
+        begin
+            dropw
+
+            # pad the stack before call
+            padw padw padw padw
+            # => [pad(16)]
+
+            exec.::miden::note::get_assets drop
+            mem_loadw
+            # => [ASSET, pad(12)]
+
+            call.::miden::contracts::faucets::basic_fungible::burn
+
+            # truncate the stack
+            dropw dropw dropw dropw
+        end
+        ";
+
+    let note = get_note_with_fungible_asset_and_script(fungible_asset, note_script);
+
+    mock_chain.add_pending_note(OutputNote::Full(note.clone()));
+    mock_chain.prove_next_block()?;
+
+    // CONSTRUCT AND EXECUTE TX (Success)
+    // --------------------------------------------------------------------------------------------
+    let executed_transaction = mock_chain
+        .build_tx_context(faucet.id(), &[note.id()], &[])?
+        .build()?
+        .execute_blocking()?;
+
+    // Prove, serialize/deserialize and verify the transaction
+    prove_and_verify_transaction(executed_transaction.clone())?;
+
+    assert_eq!(executed_transaction.account_delta().nonce_delta(), Felt::new(1));
+    assert_eq!(executed_transaction.input_notes().get_note(0).id(), note.id());
+    Ok(())
+}
+
 // TESTS BURN FUNGIBLE ASSET
 // ================================================================================================
 
+// TODO: This test uses add_existing_faucet() which creates old-style faucet accounts.
+// When the old faucet is deprecated, this test should be removed and replaced with
+// tests using create_new_faucet() or create_basic_fungible_faucet().
 #[test]
 fn prove_faucet_contract_burn_fungible_asset_succeeds() -> anyhow::Result<()> {
```

Comment:
I'm not sure what old-style faucet refers to, but if this means "existing faucets" or "non-new" ones, then we don't have to remove or replace those, from what I can tell. The difference is just between transactions creating a new faucet, which will validate a bunch of things, like that its account ID is valid or that it doesn't already contain storage that would indicate it has already issued assets, etc, and existing faucets, which don't need this validation. In tests it can be convenient to use existing faucets (or generally accounts).